### ISSUE TITLE
Curly Braces

### ISSUE THREAD

**Issue by markavitale** [microsoft/objc-guide#8]
# Convention
Use curly braces for single line scope statements, such as `if` and `for` statements

# Rationale
Adding curly braces for single line logic statements prevents errors in maintenance and refactoring over time. These types of refactoring issues around single-line if statements being refactored for multiple lines have caused real-world regressions.

# Example
https://www.imperialviolet.org/2014/02/22/applebug.html

**Comment by KranthiGV** [comment]
Good first issue? What exactly do we want from a contributer here? I would love to help out.

**Comment by markavitale** [comment]
Hey @KranthiGV, great to hear that you're interested in helping! There are many ways to contribute to the repo, the easiest of which is probably to just try and follow the guidelines in your projects and let us know when the conventions aren't working well or are particularly hard to follow.

For this particular issue, there are two main ways to contribute: 
- If you disagree with the proposed convention, join the conversation and ideally provide cases where avoiding curly braces can improve the usability/performance/maintainability of the code.
- Make a pull request for a new page that matches the format and tone of the existing pages in the guidelines. See #2 as an example pull request adding a particular convention.

For this proposed convention, I think it would most naturally fit in a new page under the Syntax section. The pull request would include:

- Adding the new page in Syntax/Foo.md, using the proposed convention and rationale as a starting point, but focusing more on getting the wording correct and broadly applicable.
    - Should the page name be Curly braces? Does "Scope" make more sense?
    - Is there a better way to generalize the convention? Is it more correct to say "avoid single line scope"?
    - The examples should be fleshed out, instead of just a link to an example of where this type of issue has occurred in the past, we can try to make a short representative code sample that makes the mistake we're avoiding more obvious
- Adding the new page to the Syntax.md table of contents
- Adding the new page to the README table of contents

**Comment by markavitale** [comment]
@KranthiGV having written all of that out, I realize that our story for how new conventions get reviewed and added to the repository is not well documented. I'll remove the good first issue label for the time being until we get that better understood ourselves and better documented.
